Enable CPU nightly performance benchmark and its Markdown report (#18444)
vllm-project/vllm: .buildkite/nightly-benchmarks/README.md

- A PR being merged into vllm.
- Every commit for those PRs with `perf-benchmarks` label AND `ready` label.

Manually Trigger the benchmark

```bash
bash .buildkite/nightly-benchmarks/scripts/run-performance-benchmarks.sh
```

Runtime environment variables:
- `ON_CPU`: set the value to '1' on Intel® Xeon® Processors. Default value is 0.
- `SERVING_JSON`: assign a json file instead of the default one for serving tests. Default value is empty string.
- `LATENCY_JSON`: assign a json file instead of the default one for latency tests. Default value is empty string.
- `THROUGHPUT_JSON`: assign a json file instead of the default one for throughout tests. Default value is empty string.
- `REMOTE_HOST`: set the remote vLLM IP to benchmark on a remote vLLM service. Default value is empty string.
- `REMOTE_PORT`: set the remote vLLM Port to benchmark on a remote vLLM service. Default value is empty string.

Nightly benchmark will be triggered when:
- Every commit for those PRs with `perf-benchmarks` label and `nightly-benchmarks` label.

## Performance benchmark details

> See [performance-benchmarks-descriptions.md](performance-benchmarks-descriptions.md) for detailed descriptions, and use `tests/latency-tests-gpu.json`, `tests/throughput-tests-gpu.json`, `tests/serving-tests-gpu.json` to configure the test cases.

Reviewer: I also think adding ```-gpu``` suffix is not necessary as it will introduce lots of changes.

Author: changed it accordingly.

The json version of the table (together with the json version of the benchmark) will be also attached to the markdown file.
The raw benchmarking results (in the format of json files) are in the `Artifacts` tab of the benchmarking.

> The `compare-json-results.py` helps to compare benchmark results json files converted using `convert-results-json-to-markdown.py`.

Reviewer 2: Why results need to be converted to MarkDown, to do a JSON comparison???
```suggestion
The `compare-json-results.py` helps to compare benchmark results JSON files converted using `convert-results-json-to-markdown.py`.
```

Author: @eero-t  you will have both benchmark_results.json and benchmark_results.md files after you run the convert-results-json-to-markdown.py.   json-to-markdown means many json files from different tests into a single md file.

> Once users trigger the benchmark script manually, the results will be generated under benchmark/results folder along with the benchmark_results.md and benchmark_results.json.  

Reviewer 2: Format file and directory names appropriately:
```suggestion
When run, benchmark script generates results under `benchmark/results` folder, along with the `benchmark_results.md` and `benchmark_results.json`.
```

> This `compare-json-results.py` compares two benchmark_results.json files with performance ratio among Output Tput, Median TTFT and Median TPOT.

Reviewer 2: Does this mean:
```suggestion
`compare-json-results.py` compares two `benchmark_results.json` files and provides performance ratio e.g. for Output Tput, Median TTFT and Median TPOT.
```
